src/plugin/track/track.js

```diff
@@ -445,88 +420,104 @@ plugin.track.getTrackPositionAt = os.track.getTrackPositionAt;
  * @param {number} stride The stride of the coordinate array.
  * @param {Array<number>=} opt_ends The end indicies of each line in a multi-line. Undefined if not a multi-line.
  *
- * @deprecated Please use `os.track.updateCurrentLine` instead.
+ * @deprecated Please use `osTrack.updateCurrentLine` instead.
  */
-plugin.track.updateCurrentLine = os.track.updateCurrentLine;
-
+const updateCurrentLine = osTrack.updateCurrentLine;
 
 /**
  * Update the z-index for a list of tracks. Ensures the current position icon for all tracks will be displayed on top
  * of the line string for every other track passed to the function.
  *
- * @param {!Array<!ol.Feature>} tracks The track features.
+ * @param {!Array<!OlFeature>} tracks The track features.
  *
- * @deprecated Please use `os.track.updateTrackZIndex` instead.
+ * @deprecated Please use `osTrack.updateTrackZIndex` instead.
  */
-plugin.track.updateTrackZIndex = os.track.updateTrackZIndex;
-
+const updateTrackZIndex = osTrack.updateTrackZIndex;
 
 /**
  * Creates a track and adds it to the Saved Places layer.
  *
- * @param {!os.track.CreateOptions} options The options object for the track.
- * @return {os.track.TrackFeatureLike|undefined} The track feature.
+ * @param {!osTrack.CreateOptions} options The options object for the track.
+ * @return {osTrack.TrackFeatureLike|undefined} The track feature.
  */
-plugin.track.createAndAdd = function(options) {
-  var track = os.track.createTrack(options);
+let createAndAdd_ = function(options) {
+  var track = osTrack.createTrack(options);
 
   if (!track) {
     var msg = 'Track creation failed. There were no valid features/coordinates to create a track.';
-    os.alertManager.sendAlert(msg, os.alert.AlertEventSeverity.WARNING);
+    alertManager.getInstance().sendAlert(msg, AlertEventSeverity.WARNING);
     return;
   }
 
-  var trackNode = plugin.file.kml.ui.updatePlacemark({
+  var trackNode = kml.ui.updatePlacemark({
     'feature': track
   });
 
-  var rootNode = plugin.places.PlacesManager.getInstance().getPlacesRoot();
+  var rootNode = PlacesManager.getInstance().getPlacesRoot();
   if (rootNode) {
-    var cmd = new plugin.file.kml.cmd.KMLNodeAdd(trackNode, rootNode);
+    var cmd = new KMLNodeAdd(trackNode, rootNode);
     cmd.title = 'Create Track';
-    os.command.CommandProcessor.getInstance().addCommand(cmd);
+    CommandProcessor.getInstance().addCommand(cmd);
 
-    plugin.track.updateTrackSource(track);
+    updateTrackSource(track);
 
     return track;
   } else {
-    goog.log.error(plugin.track.LOGGER_, 'Unable to create track: track layer missing');
+    log.error(LOGGER_, 'Unable to create track: track layer missing');
   }
 
   return;
 };
 
+/**
+ * Creates a track and adds it to the Saved Places layer.
+ *
+ * @param {!osTrack.CreateOptions} options The options object for the track.
+ * @return {osTrack.TrackFeatureLike|undefined} The track feature.
+ */
+const createAndAdd = function(options) {
```

Comment:
Ahh I was wondering how we should handle these functions we may share with external windows. This is straightforward enough, and best of all non-breaking. I like it! 👍

Reply:
If I find more Track functions that get alternate implementations like this, I might upgrade to a `CollectionManager` or a `Registry` for the `EventTarget` powers; but this is OK for here and any downstream spots at the moment. Let me know if I should pull the trigger and upgrade to that or another pattern.

Reply:
If anything I would say put the function on `TrackManager` and share that with external windows instead. You can leave the original function intact, and make the `TrackManager` function a wrapper in support of this behavior.

Reply:
I'm not sure what you mean -- If I restore createAndAdd() in TrackPlugin to what it was and in TrackManager put...
```
createAndAdd(){ this.createAndAdd_(); }
setCreateAndAdd(f) { this.createAndAdd_ = (f || TrackPlugin.createAndAdd); } 
```
...wouldn't I then have to update everywhere that calls `TrackPlugin.createAndAdd()` to instead call `TrackManager.createAndAdd()`?

OR

If I keep createAndAdd, setCreateAndAdd, and createAndAdd_ in TrackPlugin, add setCreateAndAdd to TrackManager, and call the one on TrackPlugin, wouldn't I hthen ave to also move `TrackPlugin.updateTrackSource()` and then update everywhere that uses that? That's a much smaller list at least.

Reply:
The current solution is great because it's a non-breaking change, no issue sticking with that.

If you want to make a breaking API change for the sake of a cleaner solution, then calling `TrackManager.createAndAdd()` would be my suggestion so we no longer have to export individual functions for external windows or replace the `createAndAdd` call which feels a bit hacky. We would only have to export `TrackManager` and call any functions we need to run in the main window from that.